plugins/CuraDrive/src/Settings.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2018 Ultimaker B.V.
+from UM import i18nCatalog
+
+from cura import CuraConstants
+
+
+class Settings:
+    """
+    Keeps the application settings.
+    """
+    DRIVE_API_VERSION = 1
+    DRIVE_API_URL = "{}/cura-drive/v{}".format(CuraConstants.CuraCloudAPIRoot, str(DRIVE_API_VERSION))
+
+    AUTO_BACKUP_ENABLED_PREFERENCE_KEY = "cura_drive/auto_backup_enabled"
+    AUTO_BACKUP_LAST_DATE_PREFERENCE_KEY = "cura_drive/auto_backup_date"
+
+    I18N_CATALOG_ID = "cura"
+    I18N_CATALOG = i18nCatalog(I18N_CATALOG_ID)
+
+    MESSAGE_TITLE = I18N_CATALOG.i18nc("@info:title", "Backups"),
+
+    # Translatable messages for the entire plugin.
+    translatable_messages = {
+        
+        # Menu items.
+        "extension_menu_entry": I18N_CATALOG.i18nc("@item:inmenu", "Manage backups"),
+        
+        # Notification messages.
+        "backup_failed": I18N_CATALOG.i18nc("@info:backup_status", "There was an error while creating your backup."),
+        "uploading_backup": I18N_CATALOG.i18nc("@info:backup_status", "Uploading your backup..."),
+        "uploading_backup_success": I18N_CATALOG.i18nc("@info:backup_status", "Your backup has finished uploading."),
+        "uploading_backup_error": I18N_CATALOG.i18nc("@info:backup_status",
+                                                     "There was an error while uploading your backup."),
+        "get_backups_error": I18N_CATALOG.i18nc("@info:backup_status", "There was an error listing your backups."),
+        "backup_restore_error_message": I18N_CATALOG.i18nc("@info:backup_status",
+                                                           "There was an error trying to restore your backup.")
```

Comment:
Up to you guys to change the plugin to your liking, but I strongly disagree about it being easier to read when all the text is in-line. It makes the code lines very long and hard to spot what is actual business logic (behavior) and what is (translatable) presentation. By selecting the right dict keys here it should be easy to know from within the code what type of message it is, not to mention the fact that you can use ctrl+click in any proper IDE to go to the definition. Also this makes it easier to re-use translations if they're used in multiple places (like the message title), it's hard to keep those in sync otherwise.

Reply:
In most cases, it's actually not even doing that. It's also completely breaking with the style of the rest of Cura.

So it would then be better to just set a class variable in the cases where it needs to be re-used. We've hardly ever had issues with these translations being hard to track, so adding an extra indirection is fixing a problem by means of something that's more annoying than the actual problem.